tracing-attributes/src/lib.rs

```diff
@@ -579,3 +663,80 @@ fn instrument_err(args: &[NestedMeta]) -> bool {
         _ => false,
     })
 }
+
+// terrible machinery to make async_trait-like cases works
```

Comment:
Can we have a comment here explaining what the function _does_, as well as what it's _for_?